rotkehlchen/db/eth2.py

```diff
@@ -185,38 +181,17 @@ def get_validators_with_status(
         return result
 
     def get_active_validator_indices(self, cursor: 'DBCursor') -> set[int]:
-        """Returns the indices of the tracked validators that we know have not exited
-
-        Same issues as with the get_exited_validator_indices function as far as
-        using processed events is concerned.
-        """
+        """Returns the indices of the tracked validators that we know have not exited"""
         cursor.execute(
-            'SELECT validator_index from eth2_validators WHERE withdrawable_timestamp IS NULL '
-            'OR (? - withdrawable_timestamp < ?)',
-            (ts_now(), TEMP_MAX_WITHDRAWAL_WAIT),
+            'SELECT validator_index from eth2_validators WHERE exited_timestamp IS NULL',
         )
         result = {x[0] for x in cursor}
         return result
 
     def get_exited_validator_indices(self, cursor: 'DBCursor') -> set[int]:
-        """Returns the indices of the tracked validators that we know have exited
-
-        This used to query the eth staking events for full exit of the validator
-
-        But since https://github.com/rotki/rotki/issues/8397 we need to change strategy and
-        for now we just compare withdrawable_ts with current timetamp. If the difference
-        is "big enough" we consider it exited.
-
-        The reason is that we don't always track the withdrawal address such as for eigenlayer
-        and we still need to know that a validator exited.
-
-        This solution is hacky. As per the linked issue the actual solution will come
-        in develop/1.35.0
-        """
+        """Returns the indices of the tracked validators that we know have exite"""
         cursor.execute(
-            'SELECT validator_index FROM eth2_validators WHERE withdrawable_timestamp '
-            'IS NOT NULL AND ? - withdrawable_timestamp > ?',
-            (ts_now(), TEMP_MAX_WITHDRAWAL_WAIT),
+            'SELECT validator_index FROM eth2_validators WHERE exited timestamp IS NOT NULL',
         )
         result = {x[0] for x in cursor}
         return result
```

Comment:
```suggestion
            'SELECT validator_index FROM eth2_validators WHERE exited timestamp IS NOT NULL',
        )
       return {x[0] for x in cursor}
```

Reply:
actually... it was revealed to my while praying that you forgot to do a `SELECT DISTINCT` in this query for better performance

Reply:
this query has an actual mistake ;)

There is quite a few things missing from the PR which I am also fixing.

But why would distinct make a difference? validator_index is always unique.